src/extension.ts

```diff
@@ -28,3 +31,38 @@ export async function deactivate(): Promise<void> {
     return client.stop();
   }
 }
+
+async function activateGemOutdatedButton(context: vscode.ExtensionContext) {
+  const commandId = "updateOutdatedGem";
+
+  const gemOutdatedButton = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Left,
+    0
+  );
+
+  gemOutdatedButton.text = "$(plug) Update Ruby LSP";
+  gemOutdatedButton.tooltip =
+    "The Ruby LSP gem is not up-to-date. Press this button to install the latest version of the gem.";
+  gemOutdatedButton.command = commandId;
+
+  context.subscriptions.push(
+    vscode.commands.registerCommand(commandId, async () => {
+      await updateGem();
+      await updateGemOutdatedButton(gemOutdatedButton);
+    })
+  );
+
+  updateGemOutdatedButton(gemOutdatedButton);
```

Comment:
@vinistock I know you mentioned inlining these methods but I really didn't want to write out the exact same code twice, so I'm passing the button to the method instead of having it be a global variable. Let me know if this works okay for you.